webdriver/tests/bidi/network/__init__.py

```diff
@@ -34,7 +34,7 @@ def assert_headers(event_headers, expected_headers):
     # are included in the request's headers.
     assert len(event_headers) >= len(expected_headers)
     for header in expected_headers:
-        assert next(h for h in event_headers if header == h) is not None
+        assert any(header == h for h in event_headers)
```

Comment:
So we only do a plain comparison here without `.startswith()` or similar. So couldn't it then just be the following?

```suggestion
        assert all(header in event_headers for header in expected_headers)
```

Reply:
Yeah, I wondered whether to do something more dramatic of a change here. I considered replacing the entire function with something like:

```python
event_headers_set = set(event_headers)
expected_headers_set = set(expected_headers)
assert event_headers_set <= expected_headers_set
```

In no small part because that will give better error messages from pytest.

But, from recollection, I ran into some problem whereby that caused tests to start failing. Maybe it's worthwhile trying to figure out why that wasn't working?

Reply:
@gsnedders please apply the proposal from Julian and we can see how it works. Most likely that was the problem why tests were failing for you. Those three lines look indeed way better, and I'm all for better assertion messages. One suggestion through:

> assert event_headers_set >= expected_headers_set, "Received all expected headers"

This would make it easier to classify bugs based on the assertion message.